### DISCUSSION TITLE
nginx: Why Lua VM workers consuming a lot of memory in kong 2.7.1 and Kong 2.8.1, because of that server gets down

### DISCUSSION THREAD

**Discussion by chaitalee-jadhav** [Kong/kong#9716] [Help] [answered]
Kong version ($ kong version)
2.7.1

Current Behavior
We are upgrading to Kong 2.7.1 from Kong 2.0.3 and when we are doing that experiencing out of memory error.
We never experienced this issue with 2.0.3 and it has been running perfectly for a long time now.
But right after we started upgrading, started seeing this issue and had to revert back.

We are using C5.2xlarge instances.

tps per instance: 130 request per sec 

Graph collected for lua VM wokers for a particular instance
each of the workers consuming memory in GBs
<img width="772" alt="kongupworker" src="https://user-images.githubusercontent.com/93966659/200479536-40722342-f965-4e88-9815-7f1bd27aa65f.png">

Similar issue faced with the 2.8.1

**Comment by guanlan** [comment, marked as answer]
Update to the lastest 3.x and retry please, we have fixed few memory issues.
